components/concept/engine/src/main/java/mozilla/components/concept/engine/EngineSessionState.kt

```diff
@@ -18,5 +19,14 @@ interface EngineSessionState {
      * When reading JSON from disk [Engine.createSessionState] can be used to turn it back into an [EngineSessionState]
      * instance.
      */
+    @Deprecated("Use writeTo() instead.")
```

Comment:
nit: Should we also include when this will be deleted?

Reply:
I didn't want to have a fixed date for removing this since we first need time to migrate some of our own components code over. My primary goal was to avoid any new code using this. :)
But I filed a new issue for eventually migrating everything and will also link this in a comment: https://github.com/mozilla-mobile/android-components/issues/8370